modules/spring-config/src/main/java/org/mule/runtime/config/internal/MuleArtifactContext.java

```diff
@@ -466,9 +466,9 @@ private void registerErrorMappings(SpringComponentModel componentModel, Set<Stri
         ComponentIdentifier source = parameters.containsKey(SOURCE_TYPE)
             ? buildFromStringRepresentation(parameters.get(SOURCE_TYPE)) : ANY;
 
-        muleContext.getErrorTypeRepository()
-            .lookupErrorType(source)
-            .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not find error '%s'.", source)));
+        if (!muleContext.getErrorTypeRepository().lookupErrorType(source).isPresent()) {
```

Comment:
Why is this? I think that the functional approach is better.

Reply:
https://sonarqube.kbuild.msap.io/coding_rules?open=squid%3AS2201&rule_key=squid%3AS2201

Reply:
'Return values from functions without side effects should not be ignored'
When the call to a function doesn't have any side effects, what is the point of making the call if the results are ignored? In such case, either the function call is useless and should be dropped or the source code doesn't behave as expected.

Reply:
There is another way to still get the functional approach but makes code more unreadable.

Reply:
ok, make sense.